opal/mca/event/libevent2022/libevent/select.c

```diff
@@ -166,12 +167,30 @@ select_dispatch(struct event_base *base, struct timeval *tv)
 	check_selectop(sop);
 
 	if (res == -1) {
-		if (errno != EINTR) {
-			event_warn("select");
-			return (-1);
+		if (errno == EINTR) {
+		        return (0);
 		}
-
-		return (0);
+                /* There seems to be a very subtle race condition between the
```

Comment:
Are these changes related to ULFM?

Reply:
Yes, the identifying case was in error scenarios. 

There is an upstream PR that is supposed to get merged soon with the same set of changes. 

Similar situation with the LOCK/UNLOCK on endpoint status in TCP.